msteams-platform/resources/schema/manifest-schema-dev-preview.md

```diff
@@ -991,6 +991,31 @@ The `extensions.runtimes` array configures the sets of runtimes and actions that
+|`customFunctions.functions.result`| Object | | ✔️ | Defines the type of information that is returned by the function. |
+|`customFunctions.functions.result.dimensionality`| String enum | | | Must be either `scalar` (a nonarray value) or `matrix` (a 2-dimensional array).|
+|`customFunctions.functions.stream`| Boolean | | | Default value is`false`. If `true`, the function can output repeatedly to the cell even when invoked only once. This option is useful for rapidly changing data sources, such as a stock price. The function should have no return statement. Instead, the result value is passed as the argument of the `StreamingInvocation.setResult` callback function. |
+|`customFunctions.functions.volatile`| Boolean | | | Default value is `false`. If `true`, the function recalculates each time Excel recalculates, instead of only when the formula's dependent values have changed. A function can't use both the stream and volatile properties. If the stream and volatile properties are both set to `true`, the volatile property is ignored.|
+|`customFunctions.functions.cancelable`| Boolean | | | Default value is `false`. If `true`, Excel calls the `CancelableInvocation` handler whenever the user takes an action that has the effect of canceling the function; for example, manually triggering recalculation or editing a cell that is referenced by the function. Cancelable functions are typically only used for asynchronous functions that return a single result and need to handle the cancellation of a request for data. A function can't use both the stream and cancelable properties.|
+|`customFunctions.functions.requiresAddress`| Boolean | | | Default value is `false`. If `true`, your custom function can access the address of the cell that invoked it. The `address` property of the invocation parameter contains the address of the cell that invoked your custom function. A function can't use both the `stream` and `requiresAddress` properties. |
+|`customFunctions.functions.requiresParameterAddress`| Boolean | | | Default value is `false`. If `true`, your custom function can access the addresses of the function's input parameters. This property must be used in combination with the `dimensionality` property of the `result` object, and `dimensionality` must be set to `matrix`. |
+|`customFunctions.namespace`| Object | | ✔️ | Defines the namespace for your custom functions. A namespace prepends itself to your custom functions to help customers identify your functions as part of your add-in. |
+|`customFunctions.namespace.id`| String | 32 characters | ✔️ | Specifies a unique ID for the namespace. The first character must be a letter. The subsequent characters must be alphanumeric. This value can't be localized. |
+|`customFunctions.namespace.name`| String | 32 characters | ✔️ | Specifies a name for the namespace. The first character must be a letter. The subsequent characters must be alphanumeric. This value can be localized. |
```

Comment:
If it's a requirement or a convention that this value be ALL CAPS, we should probably say that here.

Reply:
Thanks! Added "Use uppercase letters as a best practice." to customFunctions.namespace.id and customFunnctions.namespace.name.